data_tools/get.py

```diff
@@ -1,45 +1,67 @@
+import argparse
+import warnings
+
+from dateutil.parser import isoparse
+
 import firebase_client_wrapper as fcw
 
 import json
-import sys
 
-if (len(sys.argv) < 2 or len(sys.argv) > 4):
-    print ("Usage python get.py crypto_token [datasetid] [all, users, messages, schemes]")
-    exit(1)
+parser = argparse.ArgumentParser(description="Gets data from Coda's Firestore")
 
-CONTENT_TYPE = "all"
+parser.add_argument("--previous-export-file-path",
```

Comment:
I think we should make this a compulsory argument and check whether the file exists. If it does, proceed with the logic else fetch all messages. This will allow us to run new pipelines easily without having to remember to switch back to this mode after initial pipeline run.

Reply:
I don't think we need to remember. I would propose updating the pipelines to something like:
```bash
    if [ -e "$FILE" ]; then
        echo "Getting messages data from ${DATASET} (incremental update)..."
        MESSAGES=$(pipenv run python get.py --previous-export-file-path "$FILE" "$AUTH" "${DATASET}" messages)
        echo "$MESSAGES" >"$FILE"
    else
        echo "Getting messages data from ${DATASET} (full download)..."
        pipenv run python get.py "$AUTH" "${DATASET}" messages >"$FILE"
    fi
```

The reason for not making the argument compulsory is so that existing workflows are unchanged, and so we don't have to provide a dummy argument when using the data tools manually outside of the pipelines

Reply:
Makes sense thankyou.